File: app/src/main/java/in/shaaan/pcipharmd/SixthYear.java
package in.shaaan.pcipharmd;

import static in.shaaan.pcipharmd.AdUtil.loadNativeAd;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.ads.nativead.NativeAdView;
import com.google.android.material.snackbar.Snackbar;

import java.util.HashMap;
import java.util.Map;

import in.shaaan.pcipharmd.databinding.ActivitySixthYearBinding;
import in.shaaan.pcipharmd.databinding.ContentSixthYearBinding;

public class SixthYear extends AppCompatActivity implements View.OnClickListener {
    ContentSixthYearBinding yearBinding;
    AdUtil adUtil = new AdUtil();
    NativeAdView frameLayout;
    private ActivitySixthYearBinding activitySixthYearBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activitySixthYearBinding = ActivitySixthYearBinding.inflate(getLayoutInflater());
        setContentView(activitySixthYearBinding.getRoot());
        setSupportActionBar(activitySixthYearBinding.toolbar);
        yearBinding = activitySixthYearBinding.layout6y;
        adUtil.loadInterAd(this);
        yearBinding.internActivities.setOnClickListener(this);
        yearBinding.internDocuments.setOnClickListener(this);

        initializeUI();
        refreshAd();
    }

    private void initializeUI() {
        yearBinding.internActivities.setOnClickListener(this);
        yearBinding.internDocuments.setOnClickListener(this);

        activitySixthYearBinding.fab.setOnClickListener(view -> Snackbar.make(view, "Like the app? Rate it on Play Store!", Snackbar.LENGTH_LONG)
                .setAction("RATE", v -> {
                    Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=in.shaaan.pcipharmd"));
                    startActivity(intent);
                }).show());

        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
    }

    public void refreshAd() {
        loadNativeAd(this, yearBinding.nativeAd61, yearBinding.nativeCard61);
        loadNativeAd(this, yearBinding.nativeAd62, yearBinding.nativeCard62);
    }

    @Override
    public void onClick(View view) {
        Map<Integer, String> urlMap = new HashMap<>();
        urlMap.put(R.id.intern_activities, "https://shaaan.github.io/pcipd/syllabus6");
        urlMap.put(R.id.intern_documents, "https://shaaan.github.io/pcipd/syllabus6_1/");

        String url = urlMap.get(view.getId());
        if (url != null) {
            adUtil.showInterAd(this, url);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_home, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            startActivity(new Intent(this, About.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        AdUtil.interstitialAd = null;
    }
}

File: app/src/main/java/in/shaaan/pcipharmd/SixthYear.kt
package `in`.shaaan.pcipharmd

import android.view.View
import androidx.appcompat.widget.Toolbar
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.ads.nativetemplates.TemplateView
import `in`.shaaan.pcipharmd.databinding.ActivitySixthYearBinding
import `in`.shaaan.pcipharmd.databinding.ContentSixthYearBinding

/**
 * Activity for the Sixth Year syllabus.
 * What: Displays subjects for the sixth year and handles navigation to their details.
 * Why: Provides a dedicated screen for sixth-year content.
 * How: Extends `BaseYearActivity` to inherit common functionalities.
 */
class SixthYear : BaseYearActivity<ActivitySixthYearBinding, ContentSixthYearBinding>(
    ActivitySixthYearBinding::inflate
) {
    companion object {
        private const val INTERN_ACTIVITIES_PATH = "sixth_year/"
        private const val INTERN_DOCUMENTS_PATH = "internship/"
    }

    override fun getContentBinding(activityBinding: ActivitySixthYearBinding) = activityBinding.layout6y
    override fun getToolbar(activityBinding: ActivitySixthYearBinding): Toolbar = activityBinding.toolbar
    override fun getFab(activityBinding: ActivitySixthYearBinding): FloatingActionButton = activityBinding.fab

    override val subjectButtonIdToEndpointMap: Map<Int, String> = mapOf(
        R.id.intern_activities to INTERN_ACTIVITIES_PATH,
        R.id.intern_documents to INTERN_DOCUMENTS_PATH
    )

    override fun getNativeAdPlaceholderPairs(contentBinding: ContentSixthYearBinding): List<Pair<View, View>> {
        return listOf(
            Pair(contentBinding.nativeAd61, contentBinding.nativeCard61),
            Pair(contentBinding.nativeAd62, contentBinding.nativeCard62)
        )
    }

    override fun getSubjectButtons(contentBinding: ContentSixthYearBinding): List<View> {
        return listOf(
            contentBinding.internActivities, contentBinding.internDocuments
        )
    }
}
